core/src/main/java/overflowdb/storage/NodeDeserializer.java

```diff
@@ -15,136 +15,145 @@
+        return deserialize(bytes, null);
+    }
 
-    MessageUnpacker unpacker = MessagePack.newDefaultUnpacker(bytes);
-    final long id = unpacker.unpackLong();
-    final int labelStringId = unpacker.unpackInt();
-    final Object[] properties = unpackProperties(unpacker);
+    public final NodeDb deserialize(byte[] bytes, NodeRef ref) throws IOException {
+        long startTimeNanos = getStartTimeNanos();
+        if (null == bytes)
+            return null;
 
-    final String label = storage.reverseLookupStringToIntMapping(labelStringId);
-    NodeDb node = getNodeFactory(label).createNode(graph, id, ref);
-    PropertyHelper.attachProperties(node, properties);
+        MessageUnpacker unpacker = MessagePack.newDefaultUnpacker(bytes);
+        final long id = unpacker.unpackLong();
+        final int labelStringId = unpacker.unpackInt();
+        final Object[] properties = unpackProperties(unpacker);
 
-    deserializeEdges(unpacker, node, Direction.OUT);
-    deserializeEdges(unpacker, node, Direction.IN);
+        Map<Integer, String> cache = stringCache.get();
+        final String label = cache.computeIfAbsent(labelStringId,
+                idKey -> storage.reverseLookupStringToIntMapping(idKey));
+        NodeDb node = getNodeFactory(label).createNode(graph, id, ref);
+        PropertyHelper.attachProperties(node, properties);
 
-    node.markAsClean();
+        deserializeEdges(unpacker, node, Direction.OUT);
+        deserializeEdges(unpacker, node, Direction.IN);
 
-    if (statsEnabled) recordStatistics(startTimeNanos);
-    return node;
-  }
+        node.markAsClean();
 
-  private void deserializeEdges(MessageUnpacker unpacker, NodeDb node, Direction direction) throws IOException {
-    int edgeTypesCount = unpacker.unpackInt();
-    for (int edgeTypeIdx = 0; edgeTypeIdx < edgeTypesCount; edgeTypeIdx++) {
-      int edgeLabelId = unpacker.unpackInt();
-      String edgeLabel = storage.reverseLookupStringToIntMapping(edgeLabelId);
-      int edgeCount = unpacker.unpackInt();
-      for (int edgeIdx = 0; edgeIdx < edgeCount; edgeIdx++) {
-        long adjacentNodeId = unpacker.unpackLong();
-        NodeRef adjacentNode = (NodeRef) graph.node(adjacentNodeId);
-        Object[] edgeProperties = unpackProperties(unpacker);
-        node.storeAdjacentNode(direction, edgeLabel, adjacentNode, edgeProperties);
-      }
+        if (statsEnabled) recordStatistics(startTimeNanos);
+        return node;
+    }
+
+    private void deserializeEdges(MessageUnpacker unpacker, NodeDb node, Direction direction) throws IOException {
+        int edgeTypesCount = unpacker.unpackInt();
+        for (int edgeTypeIdx = 0; edgeTypeIdx < edgeTypesCount; edgeTypeIdx++) {
+            int edgeLabelId = unpacker.unpackInt();
+            Map<Integer, String> cache = stringCache.get();
+            String edgeLabel = cache.computeIfAbsent(edgeLabelId,
+                    id -> storage.reverseLookupStringToIntMapping(id));
```

Comment:
By using a cache, we reduce repeated invocations to `storage.reverseLookupStringToIntMapping`